src/SkirtBrim.cpp

```diff
@@ -223,117 +213,77 @@ std::vector<coord_t> SkirtBrim::generatePrimaryBrim(std::vector<Offset>& all_bri
-
-    Polygons ret;
-    std::vector<PolygonsPart> parts = outline.splitIntoParts();
-    for (const PolygonsPart& part : parts)
-    {
-        for (size_t hole_idx = 1; hole_idx < part.size(); hole_idx++)
-        {
-            Polygon hole_poly = part[hole_idx];
-            hole_poly.reverse();
-            Polygons disallowed_region = hole_poly.offset(10u).difference(hole_poly.offset(-line_widths_[extruder_nr] / 2 - hole_brim_distance));
-            ret = ret.unionPolygons(disallowed_region);
-        }
-    }
-    return ret;
-}
-
 coord_t SkirtBrim::generateOffset(const Offset& offset, Polygons& covered_area, std::vector<Polygons>& allowed_areas_per_extruder, SkirtBrimLine& result)
 {
     coord_t length_added;
     Polygons brim;
-    Polygons newly_covered;
+    const ExtruderConfig& extruder_config = extruders_configs_[offset.extruder_nr_];
+
+    if (std::holds_alternative<Polygons*>(offset.reference_outline_or_index_))
     {
-        if (std::holds_alternative<Polygons*>(offset.reference_outline_or_index_))
+        Polygons* reference_outline = std::get<Polygons*>(offset.reference_outline_or_index_);
+        for (ConstPolygonRef polygon : *reference_outline)
         {
-            Polygons* reference_outline = std::get<Polygons*>(offset.reference_outline_or_index_);
-            if (offset.external_only_)
-            { // prevent unioning of external polys enclosed by other parts, e.g. a small part inside a hollow cylinder.
-                for (Polygons& polys : reference_outline->sortByNesting())
-                { // offset external polygons of islands contained within another part in each batch
-                    for (PolygonRef poly : polys)
-                    {
-                        if (poly.area() < 0)
-                        {
-                            poly.reverse();
-                        }
-                    }
-                    brim.add(polys.offset(offset.offset_value_, ClipperLib::jtRound));
-                    newly_covered.add(polys.offset(offset.offset_value_ + line_widths_[offset.extruder_nr_] / 2, ClipperLib::jtRound));
-                    for (PolygonRef poly : polys)
-                    {
-                        poly.reverse();
-                    }
-                    newly_covered.add(polys); // don't remove area inside external polygon
-                }
-            }
-            else
+            const coord_t offset_value = offset.offset_value_;
+            const double area = polygon.area();
+            if ((area > 0 && offset.outside_) || (area < 0 && offset.inside_))
             {
-                brim = reference_outline->offset(offset.offset_value_, ClipperLib::jtRound);
-                newly_covered = reference_outline->offset(offset.offset_value_ + line_widths_[offset.extruder_nr_] / 2, ClipperLib::jtRound);
+                brim.add(polygon.offset(area < 0 ? -offset_value : offset_value, ClipperLib::jtRound));
             }
```

Comment:
I have a (slight) preference for
```suggestion
            const coord_t offset_value = offset.offset_value_;
            const double area = polygon.area();
            if (area > 0 && offset.outside_)
            {
                brim.add(polygon.offset(offset_value, ClipperLib::jtRound));
            }
            else if (area < 0 && offset.inside_)
            {
                brim.add(polygon.offset(-offset_value, ClipperLib::jtRound));
            }
```

Reply:
Well, that would also duplicate the first two lines, or I can place a second `if` after them to make it a bit more explicit.

Reply:
well, you did write the exact same thing here https://github.com/Ultimaker/CuraEngine/pull/2040/files#diff-696277838c64d0189f2899f11ae1efe46f331113865e2725b10fa5f8f18baef6R377-R387

Reply:
Oh no, that was in the previous code.. I can change it then